### PR TITLE
remove dead install_tar function

### PR DESCRIPTION
**Pull request by kalefranz**
@mcg1969 Care to weigh in here? I think this code is dead and should go.

### CODE REVIEW

File: conda/cli/install.py [hunk 1]
```diff
@@ -212,13 +194,6 @@ def install(args, parser, command='install'):
             raise CondaValueError("cannot mix specifications with conda package"
                                   " filenames")
 
-    # handle tar file containing conda packages
-    if len(args.packages) == 1:
-        tar_path = args.packages[0]
-        if tar_path.endswith('.tar'):
```
Comment (kalefranz) [inline conda/cli/install.py]: I think this is all dead code.  It's all completely untested at least by all of our integration tests.  I don't know what code path would actually get to it.  You'd need to `conda install my-package.tar` from the command line.  Does anybody do that?

**Comment by kalefranz** [comment]
CC @ilanschnell

**Comment by kalefranz** [comment]
Hearing no objections...

**Comment by y2kbugger** [comment]
I've used this as recently as yesterday with conda 4.2.13

To move environments to an offline PC I tar all the packages tar.bz and then install using `conda install name_of_env.tar`

see https://groups.google.com/a/continuum.io/forum/#!topic/conda/cHBYGHgvMMI
> 1) You don't have to make a conda channel
> 2) You can create a conda package out of all the stuff in your current environment (i.e. that was installed without conda)
> `conda package --pkg-name NAME --pkg-version VERSION --pkg-build BUILD`
> All the *untracked* files in your environment will end up in a conda package with the NAME-VERSION-BUILD you specify.   
> run `conda package -u `
> to see what will be in that file. 
> 3) You can build a single large .tar file that contains all the *conda-packages* you want to include
> Just copy all the conda packages into a directory and run .tar
> 4) `conda install <somefilename>.tar `
> will grab all the conda packages in the .tar file and install them. 
> 
> You can hand the .tar file around and easily install it anywhere. 

Does this mean that this will fail if I upgrade conda?
What is the alternative?

**Comment by kalefranz** [comment]
`conda package` was always labelled experimental, and never really left that stage.  conda-build is the way to *create* tarballs supported by conda.  You should be able to create a generic conda-build recipe with a [build / script](http://conda.pydata.org/docs/building/meta-yaml.html#script)  value something like

    build:
      script:  cp -R SOURCE_LOCATION $PREFIX

Once the package is built, you can invoke `conda install` with the `--use-local` flag to pull it out of the `local` conda-build channel, or use `conda install` with an absolute path to the created tarball.

As for the` --pkg-name`, `--pkg-version`, and `--pkg-build` command-line flags, I'm not sure if there's yet a way to set them dynamically with conda-build, but I know dynamic builds are on the roadmap.  If not, you might have to hard-code those in your `meta.yaml` file for now.

**Comment by y2kbugger** [comment]
Hmm not sure if that helps my use case:

I use conda completely offline and keep track each package as I install/uninstall them from specific environment. Picture ~35 pkgsomething.tar.bz2—from anaconda, conda-forge, and others channels plus the one created using `conda package`—are a folder named `nameofenv_2017-09-08`. When it comes time to put this on other PCs we tar that directory. Then, on the target PC we:

- install miniconda from exe
- `conda install nameofenv_2017-09-08.tar`

Simple deployment without need of internet.

**Comment by wherget** [comment]
I came here because I, too, wanted to use this feature, which is mentioned in some places of Anaconda's docs:
https://docs.continuum.io/anaconda/faq#how-do-i-install-packages-on-a-non-networked-machine
https://support.continuum.io/customer/en/portal/articles/2621066-conda-offline-airgap-package-installation

My use-case is redistributing a set python environment, for which this seemed appropriate. As I see it, I am currently left with requiring internet access on the target machines (and use a YAML environment descriptor), or install the tarballs (tar.bz2, that is) one-by-one. Correct?

**Pull request merged**